common/templates/xblock_v2/xblock_iframe.html

```diff
@@ -215,6 +215,16 @@
                 }
                 return url;
             },
+            notify: (eventName, data) => {
+                /** 
+                 * Used in `studio_view` to notify events and errors
+                 **/
+                if (eventName == 'save' && data.state == 'end') {
+                  window.parent.postMessage('save-end', '*');
+                } else if (eventName == 'error') {
+                  window.parent.postMessage('error', '*');
+                }
```

Comment:
What do you think about making this more generic and including `data` ? I think this will be more flexible:

```suggestion
                window.parent.postMessage({
                    type: `xblock-event`,
                    eventName,
                    data,
                }, '*');
```

Reply:
Thanks! I updated this here: https://github.com/openedx/edx-platform/pull/36029/commits/c05755cee09b3517a839c77f6858909ae17c5e58

I have used this to intercept the [cancel event](https://github.com/openedx/frontend-app-authoring/pull/1568/files#diff-1ae9e822b55474a7970d49322cb7811b863878554bd571b6a3b18f55132cebbeR28)